src/sickle.ts

```diff
@@ -224,35 +341,41 @@ class Annotator {
   }
 
   private visitProperty(className: string, p: ts.PropertyDeclaration | ts.ParameterDeclaration) {
-    let existingAnnotation = this.existingClosureAnnotation(p);
-    if (existingAnnotation) {
-      existingAnnotation += '\n';
+    let jsDoc = this.getJSDoc(p);
+    let existingAnnotation = '';
+    if (jsDoc) {
+      for (let { tagName, text } of jsDoc.tags) {
+        if (tagName === 'type') {
+          this.fail('@type not allowed in property declarations');
+        }
+        if (tagName) {
+          existingAnnotation += `@${tagName}\n`;
+        } else {
+          existingAnnotation += `${text}\n`;
+        }
+      }
     }
     this.maybeEmitJSDocType(p.type, existingAnnotation + '@type');
     this.emit(`\n    ${className}.prototype.${p.name.getText()};\n`);
   }
 
   /**
-   * Returns empty string if there is no existing annotation.
+   * Returns null if there is no existing comment.
    */
-  private existingClosureAnnotation(node: ts.Node) {
+  private getJSDoc(node: ts.Node): JSDocComment {
     let text = node.getFullText();
     let comments = ts.getLeadingCommentRanges(text, 0);
 
-    if (!comments || comments.length == 0) return '';
+    if (!comments || comments.length == 0) return null;
 
     // JS compiler only considers the last comment significant.
     let {pos, end} = comments[comments.length - 1];
     let comment = text.substring(pos, end);
-    return Annotator.getJsDocAnnotation(comment).trim();
-  }
-
-  // return empty string if comment is not JsDoc.
-  static getJsDocAnnotation(comment: string): string {
-    if (/^\/\*\*/.test(comment) && /\*\/$/.test(comment)) {
-      return comment.slice(3, comment.length - 2);
+    try {
+      return getJSDocAnnotation(comment);
+    } catch (e) {
+      this.fail(e.toString());
```

Comment:
return an empty annotation after this, so that code upstream always gets a non null value? I know this currently terminates execution, but it might not in the future (when we collect errors).